src/vfc_precexp

```diff
@@ -0,0 +1,514 @@
+			f.write("{IO}\t{ArgID}\t{Type}\t{Prec}\t{Range}\n".format(
+					IO = Arguments.at[j, 'IO'],
+					ArgID = Arguments.at[j, 'ArgID'],
+					Type = Arguments.at[j, 'Type'],
+					Prec = Arguments.at[j, 'Prec'],
+					Range = Arguments.at[j, 'Range']))
+			j = j + 1
+		arg_index = j
+
+	f.close()
+
+def Check(runPath, cmpPath, reference_dir, current_dir, config_file, env, Arguments, Operations):
+	save(Arguments, Operations, config_file)
+
+	run(runPath, current_dir, env)
+
+	return cmp(cmpPath, reference_dir, current_dir, env)
+
+def search_minimum_operations(	runPath, 
+					cmpPath, 
+					reference_dir, 
+					current_dir, 
+					config_file,
+					env, 
+					Arguments, 
+					Operations,
+					index):
+	
+	if Operations.at[index, 'Double']:
+		Operations.at[index, 'Prec64'] = 1
+
+		if not Check(runPath, cmpPath, reference_dir, current_dir, config_file, env, Arguments, Operations):
+			l , r = 1 , 52
+			flag = 1
+			while flag:
+				p = math.floor((r + l) / 2)
+				Operations.at[index, 'Prec64'] = p
+				if Check(runPath, cmpPath, reference_dir, current_dir, config_file, env, Arguments, Operations):
+					if l >= r:
+						flag = 0
+						break 
+					r = p
+				else:
+					l = p+1
+
+		Operations.at[index, 'Range64'] = 2
+
+		if not Check(runPath, cmpPath, reference_dir, current_dir, config_file, env, Arguments, Operations):
+			l , r = 2 , 11
+			flag = 1
+			while flag:
+				p = math.floor((r + l) / 2)
+				Operations.at[index, 'Range64'] = p
+				if Check(runPath, cmpPath, reference_dir, current_dir, config_file, env, Arguments, Operations):
+					if l >= r:
+						flag = 0
+						break
+					r = p
+				else:
+					l = p+1
+	else:
+		Operations.at[index, 'Prec64'] = 1
+		Operations.at[index, 'Range64'] = 2
+
+
+	if Operations.at[index, 'Float']:
+		Operations.at[index, 'Prec32'] = 1
+
+		if not Check(runPath, cmpPath, reference_dir, current_dir, config_file, env, Arguments, Operations):
+			l , r = 1 , 23
+			flag = 1
+			while flag:
+				p = math.floor((r + l) / 2)
+				Operations.at[index, 'Prec32'] = p
+				if Check(runPath, cmpPath, reference_dir, current_dir, config_file, env, Arguments, Operations):
+					if l >= r:
+						flag = 0
+						break 
+					r = p
+				else:
+					l = p+1
+
+
+
+		Operations.at[index, 'Range32'] = 2
+
+		if not Check(runPath, cmpPath, reference_dir, current_dir, config_file, env, Arguments, Operations):
+			l , r = 2 , 8
+			flag = 1
+
+			while flag:
+				p = math.floor((r + l) / 2)
+				Operations.at[index, 'Range32'] = p 
+				if Check(runPath, cmpPath, reference_dir, current_dir, config_file, env, Arguments, Operations):
+					if l >= r:
+						flag = 0
+						break
+					r = p
+				else:
+					l = p+1
+	else:
+		Operations.at[index, 'Prec32'] = 1
+		Operations.at[index, 'Range32'] = 1
+
+def search_minimum_arguments(	runPath, 
+					cmpPath, 
+					reference_dir, 
+					current_dir, 
+					config_file,
+					env, 
+					Arguments, 
+					Operations,
+					index):
+
```

Comment:
Some comments would be nice.

There is a repeated pattern here below. Could we factor the different cases ?

Reply:
I tried, tell me if the new version is better